apps/studio/src/server/modules/user/inactiveUsers.service.ts

```diff
@@ -0,0 +1,198 @@
+import { ISOMER_ADMINS_AND_MIGRATORS_EMAILS } from "~prisma/constants"
+
+import type { User } from "../database"
+import type { AccountDeactivationEmailTemplateData } from "~/features/mail/templates/types"
+import { sendAccountDeactivationEmail } from "~/features/mail/service"
+import { createBaseLogger } from "~/lib/logger"
+import { db, RoleType, sql } from "../database"
+import { PG_ERROR_CODES } from "../database/constants"
+import { MAX_DAYS_FROM_LAST_LOGIN } from "./constants"
+
+export const HOURS_IN_MS = 60 * 60 * 1000
+
+const logger = createBaseLogger({
+  path: "server/modules/user/inactiveUsers.service",
+})
+
+export function getDateOnlyInSG(daysAgo: number): Date {
+  const now = new Date()
+
+  // Convert to SG time
+  const sgOffsetMs = 8 * HOURS_IN_MS
+  const sgNow = new Date(now.getTime() + sgOffsetMs)
+
+  // Subtract daysAgo and zero out time (SG midnight)
+  sgNow.setDate(sgNow.getDate() - daysAgo)
+  sgNow.setHours(0, 0, 0, 0)
+
+  // Convert back to UTC
+  const utcDate = new Date(sgNow.getTime() - sgOffsetMs)
+  return utcDate
+}
```

Comment:
[question] concern here is readability - this seems like it's trying to get the db time and then get the last logged in via that - should we instead use another small [library](https://day.js.org/docs/en/installation/installation)? this is backend anyways, so the user evner has to load it.

Reply:
good reccommendation! https://github.com/opengovsg/isomer/pull/1489/commits/a437d6d068b25f5ca3d1246ae006a3bdda26da10